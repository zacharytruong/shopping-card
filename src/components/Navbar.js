import { Col, Container, Image, Text } from '@nextui-org/react';
import React, { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import cruiseLogo from './../media/cruiseLogo.png';

function Navbar({ orders }) {
  const [totalItems, setTotalItems] = useState(0);
  useEffect(() => {
    const tempOrders = orders.map((order) => order.quantity);
    const total = tempOrders.reduce((a, b) => parseInt(a) + parseInt(b), 0);
    setTotalItems(total);
  }, [orders]);
  return (
    <Container display="flex" wrap="wrap" css={{ h: 100 }}>
      <Col span={3}>
        <NavLink to="/">
          <Image
            width={200}
            height={100}
            src={cruiseLogo}
            alt="Ultimat World Cruise"
          />
        </NavLink>
      </Col>
      <Col
        span={8}
        css={{
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'flex-end'
        }}
      >
        <NavLink to="/shop">
          <Text h2>Shop</Text>
        </NavLink>
        <NavLink to="/checkout" style={{ display: 'flex', paddingLeft: 20 }}>
          <i
            className="fa-solid fa-cart-flatbed"
            style={{ fontSize: 35, color: '#FFFFFF' }}
          ></i>
          <Text size={35} css={{ paddingLeft: 10 }}>
            {totalItems}
          </Text>
        </NavLink>
      </Col>
    </Container>
  );
}

export default Navbar;
