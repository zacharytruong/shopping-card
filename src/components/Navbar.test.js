import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';

describe('Navbar component', () => {
  it('should render logo', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    expect(
      screen.getByRole('img', { name: /ultimat world cruise/i })
    ).toMatchSnapshot();
  });
  it('should go to homepage if user clicks on logo', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const logo = screen.getByRole('img', { name: /ultimat world cruise/i });
    userEvent.click(logo);
    expect(window.location.pathname).toMatch(/\//i);
  });
  it('should render shop menu item', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const shop = screen.getByRole('heading', { name: /shop/i });
    expect(shop).toMatchSnapshot();
  });
  it('should go to shop page if user clicks on shop menu', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const shop = screen.getByRole('heading', { name: /shop/i });
    userEvent.click(shop);
    expect(window.location.pathname).toMatch(/\/shop/i);
  });
  it('should render cart icon', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const cart = screen.getByRole('menuitem', { name: /carticon/i });
    expect(cart).toMatchSnapshot();
  });
  it('should go to checkout page if user clicks on cart icon', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const cart = screen.getByRole('menuitem', { name: /carticon/i });
    userEvent.click(cart);
    expect(window.location.pathname).toMatch(/\/checkout/i);
  });
  it('should render total number of orders', () => {
    const fakedOrders = [
      {
        quantity: 1
      },
      {
        quantity: 3
      }
    ];
    render(
      <BrowserRouter>
        <Navbar orders={fakedOrders} />
      </BrowserRouter>
    );
    const ordersQuantity = screen.getByRole('menuitem', {
      name: /ordersQuantity/i
    });
    expect(ordersQuantity.textContent).toMatch(/4/i);
  });
});
