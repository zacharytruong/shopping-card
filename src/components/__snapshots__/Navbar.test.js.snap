// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render cart icon 1`] = `
<i
  aria-label="cartIcon"
  class="fa-solid fa-cart-flatbed"
  role="menuitem"
  style="font-size: 35px; color: rgb(255, 255, 255);"
/>
`;

exports[`Navbar component should render logo 1`] = `
<img
  alt="Ultimat World Cruise"
  class="nextui-c-kbhVdb nextui-c-kbhVdb-ikZHsFe-css nextui-image"
  data-state="loading"
  height="100"
  src="cruiseLogo.png"
  width="200"
/>
`;

exports[`Navbar component should render shop menu item 1`] = `
<h2
  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css"
>
  Shop
</h2>
`;
